pkg/platformconfig/types.go

```diff
@@ -170,17 +170,18 @@ type PlatformKubeConfig struct {
 	KubeConfigPath string `json:"kubeConfigPath,omitempty"`
 
 	// TODO: Move IngressConfig here
-	DefaultServiceType               corev1.ServiceType      `json:"defaultServiceType,omitempty"`
-	DefaultFunctionNodeSelector      map[string]string       `json:"defaultFunctionNodeSelector,omitempty"`
-	DefaultHTTPIngressHostTemplate   string                  `json:"defaultHTTPIngressHostTemplate,omitempty"`
-	DefaultHTTPIngressAnnotations    map[string]string       `json:"defaultHTTPIngressAnnotations,omitempty"`
-	DefaultFunctionPriorityClassName string                  `json:"defaultFunctionPriorityClassName,omitempty"`
-	DefaultFunctionServiceAccount    string                  `json:"defaultFunctionServiceAccount,omitempty"`
-	ValidFunctionPriorityClassNames  []string                `json:"validFunctionPriorityClassNames,omitempty"`
-	DefaultFunctionPodResources      PodResourceRequirements `json:"defaultFunctionPodResources,omitempty"`
-	DefaultSidecarResources          PodResourceRequirements `json:"defaultSidecarResources,omitempty"`
-	DefaultFunctionTolerations       []corev1.Toleration     `json:"defaultFunctionTolerations,omitempty"`
-	PreemptibleNodes                 *PreemptibleNodes       `json:"preemptibleNodes,omitempty"`
+	DefaultServiceType                   corev1.ServiceType      `json:"defaultServiceType,omitempty"`
+	DefaultFunctionNodeSelector          map[string]string       `json:"defaultFunctionNodeSelector,omitempty"`
+	DefaultHTTPIngressHostTemplate       string                  `json:"defaultHTTPIngressHostTemplate,omitempty"`
+	DefaultHTTPIngressAnnotations        map[string]string       `json:"defaultHTTPIngressAnnotations,omitempty"`
+	DefaultFunctionPriorityClassName     string                  `json:"defaultFunctionPriorityClassName,omitempty"`
+	DefaultFunctionServiceAccount        string                  `json:"defaultFunctionServiceAccount,omitempty"`
+	ValidFunctionPriorityClassNames      []string                `json:"validFunctionPriorityClassNames,omitempty"`
+	DefaultFunctionPodResources          PodResourceRequirements `json:"defaultFunctionPodResources,omitempty"`
+	DefaultSidecarResources              PodResourceRequirements `json:"defaultSidecarResources,omitempty"`
+	DefaultFunctionTolerations           []corev1.Toleration     `json:"defaultFunctionTolerations,omitempty"`
+	PreemptibleNodes                     *PreemptibleNodes       `json:"preemptibleNodes,omitempty"`
+	MergePlatformAndProjectNodeSelectors bool                    `json:"mergePlatformAndProjectNodeSelectors"`
```

Comment:
Since this defaults to false, it will break BC.
If we want to change behaviors only in IGZ systems. we can default this to true (i.e. change to `skip` or something), and change the platform configurations in IGZ (-> provazio).
This will achieve it without breaking Nuclio BC.

Reply:
@TomerShor great idea! Will add subtask for provazio to Jira.

Reply:
https://iguazio.atlassian.net/browse/NUC-173